packages/anywidget/src/widget.js

```diff
@@ -309,53 +309,58 @@ class Runtime {
 	/** @type {import('solid-js').Resource<Result<AnyWidget & { url: string }>>} */
 	// @ts-expect-error - Set synchronously in constructor.
 	#widget_result;
+	/** @type {Promise<void>} */
+	model_created;
 
 	/** @param {base.DOMWidgetModel} model */
 	constructor(model) {
-		this.#disposer = solid.createRoot((dispose) => {
-			let [css, set_css] = solid.createSignal(model.get("_css"));
-			model.on("change:_css", () => {
-				let id = model.get("_anywidget_id");
-				console.debug(`[anywidget] css hot updated: ${id}`);
-				set_css(model.get("_css"));
-			});
-			solid.createEffect(() => {
-				let id = model.get("_anywidget_id");
-				load_css(css(), id);
-			});
-
-			/** @type {import("solid-js").Signal<string>} */
-			let [esm, setEsm] = solid.createSignal(model.get("_esm"));
-			model.on("change:_esm", async () => {
-				let id = model.get("_anywidget_id");
-				console.debug(`[anywidget] esm hot updated: ${id}`);
-				setEsm(model.get("_esm"));
+		this.model_created = new Promise((resolve) => {
```

Comment:
May I suggest using [`Promise.withResolvers`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/withResolvers) to help with readability here?

```ts
let { promise, resolve } = Promise.withResolvers();
this.ready = promise;
this.#disposer = solid.createRoot((dispose) => {
  /* ... */ 
  resolve();
  /* ... */ 
});
```